src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -1444,4 +1454,42 @@ protected function get_edit_media_item_args() {
 		);
 	}
 
+	/**
+	 * Issue an incorrect-usage error when a featured image has been assigned to an attachment with
+	 * a MIME type that doesn't support post thumbnails.
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/41692
+	 *
+	 * @param int    $attachment_id Attachment ID.
+	 * @param string $method        Method name.
+	 */
+	private function maybe_did_featured_media_wrong( $attachment_id, $method ) {
```

Comment:
What does this method do?

Reply:
This method is responsible for determining whether a created or updated attachment supports featured images. If an attachment is created or updated via the API with a featured image, but the attachment has a MIME type that doesn't support featured images, this method sends a `_doing_it_wrong()`.

Reply:
Doing it wrong is not how the REST API handles errors. I don't think this method is a good idea.